web/packages/teleport/src/Discover/useDiscover.tsx

```diff
@@ -93,6 +94,14 @@ type DiscoverProviderProps = {
   mockCtx?: DiscoverContextState;
   // Extra view configs that are passed in. This is used to add view configs from Enterprise.
   eViewConfigs?: EViewConfigs;
+  // isUpdateFlow indicates if the Discover flow is an update resource flow.
+  isUpdateFlow?: boolean;
+  // resourceSpecForUpdate specifies ResourceSpec which should be used to
+  // start a Discover flow.
+  resourceSpecForUpdate?: ResourceSpec;
+  // agentMetaForUpdate includes data that will be used to prepopulate input fields
+  // in the respective Discover compnents.
+  agentMetaForUpdate?: AgentMeta;
```

Comment:
I think we can model this more strictly to avoid invalid states. For example, now you can set only one of these fields and I'm not sure if the component will handle it correctly.
It would be better if it looked like this:
```ts
  //updateFlow indicates if the Discover flow is an update resource flow
  updateFlow?: {
    // resourceSpec specifies ResourceSpec which should be used to
    // start a Discover flow.
    resourceSpec: ResourceSpec;
    // agentMeta includes data that will be used to prepopulate input fields
    // in the respective Discover compnents.
    agentMeta: AgentMeta;
}
```

What do you think?

Reply:
I like it. Thanks. Updated in https://github.com/gravitational/teleport/pull/43774/commits/c72e310604260f6fab6e5b6f0c06354a84dbb590